app/controllers/admin/requests_controller.rb

```diff
@@ -0,0 +1,13 @@
+class Admin::RequestsController < AdminController
+  def index
+    @requests = Request.order_day_begin.includes(:user, :books)
+  end
+
+  def show
+    @request = Request.find_by id: params[:id]
+    return if @request
+
+    flash[:danger] = t ".request_invalid"
+    redirect_to root_url
```

Comment:
Nếu ko tìm thấy bản ghi thì nên chuyển hướng về trang index tương ứng em nhé
user mất thêm thao tác để quay về index nếu cho chuyển hướng về root em à
Fix tương tự các action bên dưới
```suggestion
    redirect_to admin_requests_path
```